src/conn.rs

```diff
@@ -824,17 +791,33 @@ impl<const N: usize, P: ConnectionPeer> Connection<N, P> {
             self.readable.notify_one();
         }
 
-        // If both endpoints have exchanged FINs and all data has been received/acknowledged, then
-        // close the connection.
+        // One way Fin case 1, we are sending data, we send our fin and if it is acked, the reiecever got all our data and we close
```

Comment:
`// One-way Fin: Case 1 - We have sent our data and then our fin. If this fin is acked, then the receiver got all our data and we close`